Swabbr.Api/Controllers/ReactionsController.cs

```diff
@@ -191,103 +187,35 @@ public async Task<IActionResult> GetReactionCountForVlog([FromRoute] Guid vlogId
-        {
-            if (reactionId.IsNullOrEmpty()) { return BadRequest(this.Error(ErrorCodes.InvalidInput, "Reaction id can't be null")); }
+                // Map.
+                var result = MapperReaction.Map(postedReaction);
 
-            try
-            {
-                var user = await _userManager.GetUserAsync(User).ConfigureAwait(false);
-                return Ok(new ReactionUploadUrlOutputModel
-                {
-                    UploadUri = await _reactionService.GetNewUploadUriAsync(user.Id, reactionId).ConfigureAwait(false)
-                });
+                // Return.
+                return Ok(result);
             }
             catch (EntityNotFoundException e)
             {
                 logger.LogError(e.Message);
-                return Conflict(this.Error(ErrorCodes.EntityNotFound, "Could not find reaction"));
-            }
-            catch (ReactionStateException e)
-            {
-                logger.LogError(e.Message);
-                return Conflict(this.Error(ErrorCodes.InvalidOperation, "Reaction not in upload awaiting state"));
-            }
-            catch (Exception e)
-            {
-                logger.LogError(e.Message);
-                return Conflict(this.Error(ErrorCodes.InvalidOperation, "Could not post new reaction"));
-            }
-        }
-
-        /// <summary>
-        /// Should be called when the user finishes uploading a <see cref="Reaction"/>.
-        /// </summary>
-        /// <param name="reactionId">Internal <see cref="Core.Entities.Reaction"/> id</param>
-        /// <returns><see cref="OkObjectResult"/> with <see cref="ReactionOutputModel"/></returns>
-        [HttpPost("finished_uploading")]
-        [ProducesResponseType((int)HttpStatusCode.OK)]
-        [ProducesResponseType((int)HttpStatusCode.BadRequest)]
-        [ProducesResponseType((int)HttpStatusCode.Conflict)]
-        public async Task<IActionResult> OnFinishedUploading(Guid reactionId)
-        {
-            if (reactionId.IsNullOrEmpty()) { return BadRequest(this.Error(ErrorCodes.InvalidInput, "Reaction id can't be null")); }
-
-            try
-            {
-                var user = await _userManager.GetUserAsync(User).ConfigureAwait(false);
-                await _reactionService.OnFinishedUploadingReactionAsync(reactionId).ConfigureAwait(false);
-                return Ok();
-            }
-            catch (EntityNotFoundException e)
-            {
-                logger.LogError(e.Message);
-                return Conflict(this.Error(ErrorCodes.EntityNotFound, "Could not find reaction"));
-            }
-            catch (ReactionStateException e)
-            {
-                logger.LogError(e.Message);
-                return Conflict(this.Error(ErrorCodes.InvalidOperation, "Reaction not in upload awaiting state"));
+                return Conflict(this.Error(ErrorCodes.EntityNotFound, "Could not find target vlog"));
```

Comment:
Maybe return `problem()` ?

Reply:
We need consistent error handling, created #195. This is one for a separate PR in which I also take the controllers into account.